subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r43/ToolingApiUnsupportedConsumerCrossVersionSpec.groovy

```diff
@@ -14,96 +14,60 @@
  * limitations under the License.
  */
 
-package org.gradle.integtests.tooling
+package org.gradle.integtests.tooling.r43
 
-import org.gradle.integtests.tooling.r18.NullAction
-import org.gradle.integtests.tooling.fixture.TargetGradleVersion
-import org.gradle.integtests.tooling.fixture.ToolingApiSpecification
 import org.gradle.integtests.tooling.fixture.ToolingApiVersion
-import org.gradle.tooling.ProjectConnection
-import org.gradle.tooling.model.eclipse.EclipseProject
 import org.gradle.util.GradleVersion
 
-class ToolingApiUnsupportedVersionCrossVersionSpec extends ToolingApiSpecification {
-    def setup() {
-        file("build.gradle") << """
-task noop {
-    doLast {
-        println "noop"
-    }
-}
-"""
-    }
-
+@ToolingApiVersion("current")
+class ToolingApiUnsupportedConsumerCrossVersionSpec extends ToolingApiVersionSpecification {
     @ToolingApiVersion("<1.2")
-    @TargetGradleVersion(">=2.0")
     def "provider rejects build request from a tooling API client older than 1.2"() {
         when:
-        withConnection { ProjectConnection connection ->
-            def build = connection.newBuild()
-            build.forTasks("noop")
-            build.run()
-        }
+        build()
 
         then:
         caughtGradleConnectionException = thrown()
-        caughtGradleConnectionException.cause.message.contains('Support for clients using a tooling API version older than 2.0 was removed in Gradle 3.0. You should upgrade your tooling API client to version 2.0 or later.')
+        caughtGradleConnectionException.cause.message.contains('Support for clients using a tooling API version older than 2.0 was removed in Gradle 3.0. You should upgrade your tooling API client to version 3.0 or later.')
     }
 
     @ToolingApiVersion(">=1.2 <2.0")
-    @TargetGradleVersion(">=3.0")
```

Comment:
❌ We can't remove these constraints, as the message was different in older versions. This would fail when running the full range of versions. Instead, we should create a new test with the current expectations and limit this one to <=Gradle 4.2. Alternatively we can adjust the helper methods (like "consumerDeprecationMessage") to return the correct expected message for each version.